src/Jupyter/CustomShell/ClientInfoHandler.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Extensions.Logging;
+using Microsoft.Jupyter.Core;
+using Microsoft.Jupyter.Core.Protocol;
+using Newtonsoft.Json;
+
+namespace Microsoft.Quantum.IQSharp.Jupyter
+{
+
+    /// <summary>
+    ///     Represents information returned to the client about the current
+    ///     kernel instance, such as the current hosting environment.
+    /// </summary>
+    public class ClientInfoContent : MessageContent
+    {
+        [JsonProperty("hosting_environment")]
```

Comment:
why are we using this naming convention ("hosting_environment") for a json property instead of camel casing ("hostingEnvironment") which is the standard for JS/JSON properties (https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers)?

Reply:
Thanks for pointing that out! My thinking here was that Jupyter messaging protocol follows `snake_case` as per its Python heritage, rather than the more conventional JSON usage of `camelCase`. Since we're already using `snake_case` for many other uses of JSON in this kernel, my thought was that it would be more consistent to intentionally violate JSON conventions in order to stay consistent with existing serialization properties for other Jupyter message contents.

Reply:
If we want this naming convention for all of our objects, maybe we could centralize it in the serialization settings, for example:

var serializerSettings = new JsonSerializerSettings()
{
    ContractResolver = new DefaultContractResolver()
    {
        NamingStrategy = new SnakeCaseNamingStrategy()
    },
};

return JsonConvert.SerializeObject(obj, serializerSettings);

Reply:
Sounds good, though I'd like to defer that to a future PR. At the moment, the call to `SerializeObject` happens in jupyter-core, not iqsharp, such that that change would require working across repos to coordinate.